dependency_test.go

```diff
@@ -0,0 +1,119 @@
+package testcontainers
+
+import (
+	"context"
+	"fmt"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"strings"
+	"testing"
+)
+
+func Test_ContainerDependsOn(t *testing.T) {
+	type TestCase struct {
+		name                string
+		configureDependants func(ctx context.Context, t *testing.T) []*ContainerDependency
+		containerRequest    ContainerRequest
+		expectedEnv         []string
+		expectedError       string
+	}
+
+	testCases := []TestCase{
+		{
+			name: "dependency is started and the dns name is passed as an environment variable",
+			configureDependants: func(ctx context.Context, t *testing.T) []*ContainerDependency {
+				nginxReq := ContainerRequest{
+					Image:        nginxAlpineImage,
+					ExposedPorts: []string{nginxDefaultPort},
+					WaitingFor:   wait.ForListeningPort(nginxDefaultPort),
+				}
+
+				return []*ContainerDependency{
+					NewContainerDependency(nginxReq, "FIRST_DEPENDENCY"),
+					NewContainerDependency(nginxReq, "SECOND_DEPENDENCY"),
```

Comment:
Here's the interface that I came up with for dependencies. I still have to add more tests and work on edge cases, but thought I'd push this up to get some eyes on it.